cli/silence_update.go

```diff
@@ -0,0 +1,147 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"github.com/prometheus/alertmanager/cli/format"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+type getResponse struct {
+	Status    string        `json:"status"`
+	Data      types.Silence `json:"data,omitempty"`
+	ErrorType string        `json:"errorType,omitempty"`
+	Error     string        `json:"error,omitempty"`
+}
+
+var updateFlags *flag.FlagSet
+var updateCmd = &cobra.Command{
+	Use:     "update <id> ...",
+	Aliases: []string{"extend"},
+	Args:    cobra.MinimumNArgs(1),
+	Short:   "Update silences",
+	Long:    `Extend or update existing silence in Alertmanager.`,
```

Comment:
Could you update this with a longer description, with a usage example? It would also be good to note that `expires-on` will override `expires`.

Reply:
I'dd add that in once we decided on the parameter names :)